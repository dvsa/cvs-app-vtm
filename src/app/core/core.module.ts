import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { GlobalErrorComponent } from './components/global-error/global-error.component';
import { SpinnerComponent } from './components/spinner/spinner.component';
import { FooterComponent } from './components/footer/footer.component';
import { HeaderComponent } from './components/header/header.component';
import { RouterModule } from '@angular/router';
import { BreadcrumbsComponent } from './components/breadcrumbs/breadcrumbs.component';
import { PageNotFoundComponent } from './components/page-not-found/page-not-found.component';
import { ServerErrorComponent } from './components/server-error/server-error.component';

@NgModule({
  declarations: [FooterComponent, HeaderComponent, GlobalErrorComponent, SpinnerComponent, BreadcrumbsComponent, PageNotFoundComponent, ServerErrorComponent],
  imports: [CommonModule, RouterModule],
  exports: [FooterComponent, HeaderComponent, GlobalErrorComponent, SpinnerComponent, BreadcrumbsComponent]
})
export class CoreModule {}
