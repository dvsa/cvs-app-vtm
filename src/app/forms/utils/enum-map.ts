import { MultiOptions } from '@forms/models/options.model';

export function getOptionsFromEnum(object: object): MultiOptions {
  return Object.values(object).map(value => ({ value, label: value.charAt(0).toUpperCase() + value.slice(1) }));
}

export function getOptionsFromEnumOneChar(object: object): MultiOptions {
  return Object.values(object).map(value => ({ value, label: value.charAt(0).toUpperCase() }));
}

export function getOptionsFromEnumAcronym(object: object): MultiOptions {
  return Object.values(object).map(value => ({ value, label: value.toUpperCase() }));
}
