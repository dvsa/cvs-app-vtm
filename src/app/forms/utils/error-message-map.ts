import { formatDate } from '@angular/common';
import { AsyncValidatorNames } from '@forms/models/async-validators.enum';
import { ValidatorNames } from '@forms/models/validators.enum';

const DEFAULT_LABEL = 'This field';
export const ErrorMessageMap: Record<string, (...args: any) => string> = {
  // Date errors
  invalidDate: (err: { error: boolean; reason: string; index: number }) => `${err.reason}`,
  invalidOption: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} is invalid`,
  invalidTestResult: (err: { message: string }) => err.message,

  [ValidatorNames.AheadOfDate]: (err: { sibling: string; date: Date }, label?: string) =>
    `${label || 'This date'} must be ahead of ${err.sibling || 'the previous date'}${err.date ? formatDate(err.date, ' (dd/MM/yyyy)', 'en') : ''}`,
  [ValidatorNames.CustomPattern]: (err: { message: string }, label?: string) => `${label || DEFAULT_LABEL} ${err.message}`,
  [ValidatorNames.DateNotExceed]: (err: { sibling: string; months: number }, label?: string) =>
    `${label || 'This date'} must be less than ${err.months || 'X'} months after ${err.sibling || 'the previous date'}`,
  [ValidatorNames.Defined]: (err: boolean, label?: string) => `${label} is required`,
  [ValidatorNames.FutureDate]: (err: boolean, label?: string) => `${label || 'This date'} must be in the future`,
  [ValidatorNames.Max]: (err: { max: number }, label?: string) => `${label || DEFAULT_LABEL} must be less than or equal to ${err.max}`,
  [ValidatorNames.MaxLength]: (err: { requiredLength: number }, label?: string) =>
    `${label || DEFAULT_LABEL} must be less than or equal to ${err.requiredLength} characters`,
  [ValidatorNames.Min]: (err: { min: number }, label?: string) => `${label || DEFAULT_LABEL} must be greater than or equal to ${err.min}`,
  [ValidatorNames.MinLength]: (err: { requiredLength: number }, label?: string) =>
    `${label || DEFAULT_LABEL} must be greater than or equal to ${err.requiredLength} characters`,
  [ValidatorNames.NotZNumber]: () => "The VRM/Trailer ID cannot be in a format that is 7 digits followed by the character 'Z'",
  [ValidatorNames.PastDate]: (err: boolean, label?: string) => `${label || 'This date'} must be in the past`,
  [ValidatorNames.Pattern]: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} must match a pattern`,
  [ValidatorNames.Required]: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} is required`,
  [ValidatorNames.RequiredIfEquals]: (err: { sibling: string }, label?: string) => `${label || DEFAULT_LABEL} is required with ${err.sibling}`,
  [ValidatorNames.ValidateDefectNotes]: () => 'Notes is required',
  [ValidatorNames.ValidateProhibitionIssued]: () => 'Prohibition notice has not been issued.',

  [AsyncValidatorNames.RequiredIfNotAbandoned]: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} is required`,
  [AsyncValidatorNames.RequiredIfNotFail]: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} is required`,
  [AsyncValidatorNames.RequiredIfNotResult]: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} is required`,
  [AsyncValidatorNames.RequiredIfNotResultAndSiblingEquals]: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} is required`
};
