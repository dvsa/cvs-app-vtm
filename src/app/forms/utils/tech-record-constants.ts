import { FormNode } from '@forms/services/dynamic-form.types';
import { CarTechRecord } from '@forms/templates/car/car-tech-record.template';
import { ApplicantDetails } from '@forms/templates/general/applicant-details.template';
import { HgvAndTrlTypeApprovalTemplate } from '@forms/templates/general/approval-type.template';
import { Audit } from '@forms/templates/general/audit.template';
import { DocumentsTemplate } from '@forms/templates/general/documents.template';
import { HgvAndTrlBodyTemplate } from '@forms/templates/general/hgv-trl-body.template';
import { LettersTemplate } from '@forms/templates/general/letters.template';
import { ManufacturerTemplate } from '@forms/templates/general/manufacturer.template';
import { NotesTemplate } from '@forms/templates/general/notes.template';
import { PlatesTemplate } from '@forms/templates/general/plates.template';
import { HgvDimensionsTemplate } from '@forms/templates/hgv/hgv-dimensions.template';
import { HgvTechRecord } from '@forms/templates/hgv/hgv-tech-record.template';
import { tyresTemplateHgv } from '@forms/templates/hgv/hgv-tyres.template';
import { HgvWeight } from '@forms/templates/hgv/hgv-weight.template';
import { LgvTechRecord } from '@forms/templates/lgv/lgv-tech-record.template';
import { MotorcycleTechRecord } from '@forms/templates/motorcycle/motorcycle-tech-record.template';
import { PsvTypeApprovalTemplate } from '@forms/templates/psv/psv-approval-type.template';
import { PsvBodyTemplate } from '@forms/templates/psv/psv-body.template';
import { PsvBrakesTemplate } from '@forms/templates/psv/psv-brakes.template';
import { PsvDdaTemplate } from '@forms/templates/psv/psv-dda.template';
import { PsvDimensionsTemplate } from '@forms/templates/psv/psv-dimensions.template';
import { PsvNotes } from '@forms/templates/psv/psv-notes.template';
import { PsvTechRecord } from '@forms/templates/psv/psv-tech-record.template';
import { PsvTyresTemplate } from '@forms/templates/psv/psv-tyres.template';
import { PsvWeightsTemplate } from '@forms/templates/psv/psv-weight.template';
import { SmallTrailerTechRecord } from '@forms/templates/small-trailer/small-trailer-tech-record.template';

import { TechRecordReasonForCreationSection } from '@forms/templates/test-records/section-templates/techRecord_reasonForCreation/techRecordReasonForCreation.template';
import { TrlAuthIntoServiceTemplate } from '@forms/templates/trl/trl-auth-into-service.template';
import { TrlBrakesTemplate } from '@forms/templates/trl/trl-brakes.template';
import { TrlDimensionsTemplate } from '@forms/templates/trl/trl-dimensions.template';
import { TrlPurchasers } from '@forms/templates/trl/trl-purchaser.template';
import { TrlTechRecordTemplate } from '@forms/templates/trl/trl-tech-record.template';
import { tyresTemplateTrl } from '@forms/templates/trl/trl-tyres.template';
import { TrlWeight } from '@forms/templates/trl/trl-weight.template';
import { VehicleTypes } from '@models/vehicle-tech-record.model';

// The map below initializes the array of sections that the *ngFor in tech summary component's template will iterate over.
// The order in which each section is introduced in the array will determine its order on the page when rendered.
// Sections which use custom components require a FormNode object with 'name' and 'label' properties.

export const vehicleTemplateMap = new Map<VehicleTypes, Array<FormNode>>([
  [
    VehicleTypes.PSV,
    [
      /*  1 */ TechRecordReasonForCreationSection,
      /*  2 */ PsvNotes,
      /*  3 */ PsvTechRecord,
      /*  4 */ PsvTypeApprovalTemplate,
      /*  5 */ PsvBrakesTemplate,
      /*  6 */ PsvDdaTemplate,
      /*  7 */ DocumentsTemplate,
      /*  8 */ PsvBodyTemplate,
      /*  9 */ PsvWeightsTemplate,
      /* 10 */ PsvTyresTemplate,
      /* 11 */ PsvDimensionsTemplate
    ]
  ],
  [
    VehicleTypes.HGV,
    [
      /*  1 */ TechRecordReasonForCreationSection,
      /*  2 */ NotesTemplate,
      /*  3 */ HgvTechRecord,
      /*  4 */ HgvAndTrlTypeApprovalTemplate,
      /*  5 */ ApplicantDetails,
      /*  6 */ DocumentsTemplate,
      /*  7 */ HgvAndTrlBodyTemplate,
      /*  8 */ HgvWeight,
      /*  9 */ tyresTemplateHgv,
      /* 10 */ HgvDimensionsTemplate,
      /* 11 */ PlatesTemplate
    ]
  ],
  [
    VehicleTypes.TRL,
    [
      /*  1 */ TechRecordReasonForCreationSection,
      /*  2 */ NotesTemplate,
      /*  3 */ TrlTechRecordTemplate,
      /*  4 */ HgvAndTrlTypeApprovalTemplate,
      /*  5 */ ApplicantDetails,
      /*  6 */ DocumentsTemplate,
      /*  7 */ LettersTemplate,
      /*  8 */ HgvAndTrlBodyTemplate,
      /*  9 */ TrlWeight,
      /* 10 */ tyresTemplateTrl,
      /* 11 */ TrlBrakesTemplate,
      /* 12 */ TrlPurchasers,
      /* 13 */ TrlDimensionsTemplate,
      /* 14 */ PlatesTemplate,
      /* 15 */ TrlAuthIntoServiceTemplate,
      /* 16 */ ManufacturerTemplate
    ]
  ],
  [
    VehicleTypes.SMALL_TRL,
    [/* 1 */ TechRecordReasonForCreationSection, /* 2 */ SmallTrailerTechRecord, /* 3 */ ApplicantDetails, /* 4 */ NotesTemplate, /* 5 */ Audit]
  ],
  [
    VehicleTypes.LGV,
    [/* 1 */ TechRecordReasonForCreationSection, /* 2 */ LgvTechRecord, /* 3 */ ApplicantDetails, /* 4 */ NotesTemplate, /* 5 */ Audit]
  ],
  [
    VehicleTypes.CAR,
    [/* 1 */ TechRecordReasonForCreationSection, /* 2 */ CarTechRecord, /* 3 */ ApplicantDetails, /* 4 */ NotesTemplate, /* 5 */ Audit]
  ],
  [
    VehicleTypes.MOTORCYCLE,
    [/* 1 */ TechRecordReasonForCreationSection, /* 2 */ MotorcycleTechRecord, /* 3 */ ApplicantDetails, /* 4 */ NotesTemplate, /* 5 */ Audit]
  ]
]);
