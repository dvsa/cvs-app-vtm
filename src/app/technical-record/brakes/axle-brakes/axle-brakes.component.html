<dl class="govuk-summary-list">
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">Axle ({{ axle.axleNumber }})</dt>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">Brake actuator</dt>
    <dd id="test-brakes-brakeActuator" class="govuk-summary-list__value">
      {{ axle?.brakes?.brakeActuator | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">Lever length</dt>
    <dd id="test-brakes-leverLength" class="govuk-summary-list__value">
      {{ axle?.brakes?.leverLength | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">Spring brake parking</dt>
    <dd id="test-brakes-springBrakeParking" class="govuk-summary-list__value">
      {{ axle?.brakes?.springBrakeParking | DefaultNullOrEmpty }}
    </dd>
    <dd class="govuk-summary-list__value"></dd>
  </div>
</dl>
