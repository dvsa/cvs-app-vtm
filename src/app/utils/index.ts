export * from './mockStore';
export * from './test-model.utils';
export * from './testing.utils';
export * from './form.utils';


