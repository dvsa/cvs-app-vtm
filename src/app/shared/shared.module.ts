import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { DocumentRetrievalService } from '@api/document-retrieval';
import { RoleRequiredDirective } from '@directives/app-role-required.directive';
import { AccordionControlComponent } from './components/accordion-control/accordion-control.component';
import { AccordionComponent } from './components/accordion/accordion.component';
import { BannerComponent } from './components/banner/banner.component';
import { BaseDialogComponent } from './components/base-dialog/base-dialog.component';
import { ButtonGroupComponent } from './components/button-group/button-group.component';
import { ButtonComponent } from './components/button/button.component';
import { IconComponent } from './components/icon/icon.component';
import { NumberPlateComponent } from './components/number-plate/number-plate.component';
import { PaginationComponent } from './components/pagination/pagination.component';
import { TagComponent } from './components/tag/tag.component';
import { TestCertificateComponent } from './components/test-certificate/test-certificate.component';
import { RetrieveDocumentDirective } from './directives/retrieve-document/retrieve-document.directive';
import { PreventDoubleClickDirective } from './directives/prevent-double-click/prevent-double-click.directive';
import { DefaultNullOrEmpty } from './pipes/default-null-or-empty/default-null-or-empty.pipe';
import { DigitGroupSeparatorPipe } from './pipes/digit-group-separator/digit-group-separator.pipe';
import { RefDataDecodePipe } from './pipes/ref-data-decode/ref-data-decode.pipe';
import { TestTypeNamePipe } from './pipes/test-type-name/test-type-name.pipe';

@NgModule({
  declarations: [
    DefaultNullOrEmpty,
    ButtonGroupComponent,
    ButtonComponent,
    BannerComponent,
    RoleRequiredDirective,
    TagComponent,
    NumberPlateComponent,
    IconComponent,
    TestTypeNamePipe,
    AccordionComponent,
    AccordionControlComponent,
    PaginationComponent,
    TestCertificateComponent,
    PreventDoubleClickDirective,
    BaseDialogComponent,
    DigitGroupSeparatorPipe,
    RefDataDecodePipe,
    RetrieveDocumentDirective
  ],
  imports: [CommonModule, RouterModule],
  exports: [
    DefaultNullOrEmpty,
    ButtonGroupComponent,
    ButtonComponent,
    BannerComponent,
    RoleRequiredDirective,
    TagComponent,
    NumberPlateComponent,
    IconComponent,
    TestTypeNamePipe,
    AccordionComponent,
    AccordionControlComponent,
    PaginationComponent,
    TestCertificateComponent,
    BaseDialogComponent,
    DigitGroupSeparatorPipe,
    RefDataDecodePipe,
    RetrieveDocumentDirective
  ],
  providers: [DocumentRetrievalService]
})
export class SharedModule {}
