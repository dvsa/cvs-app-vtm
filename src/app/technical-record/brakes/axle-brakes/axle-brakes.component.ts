import { Component, OnInit, ChangeDetectionStrategy, Input } from '@angular/core';
import { Axle } from '@app/models/tech-record.model';

@Component({
  selector: 'vtm-axle-brakes',
  templateUrl: './axle-brakes.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class AxleBrakesComponent implements OnInit {

  @Input() vehicleType: string;
  @Input() axle: Axle;
  constructor() { }

  ngOnInit() {
  }
}
