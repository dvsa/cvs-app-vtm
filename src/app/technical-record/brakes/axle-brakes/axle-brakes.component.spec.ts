import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { AxleBrakesComponent } from './axle-brakes.component';
import { SharedModule } from '@app/shared/shared.module';
import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { TESTING_UTILS } from '../../../utils/testing.utils';

describe('AxleBrakesComponent', () => {
  let component: AxleBrakesComponent;
  let fixture: ComponentFixture<AxleBrakesComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [SharedModule],
      declarations: [ AxleBrakesComponent ],
      schemas: [CUSTOM_ELEMENTS_SCHEMA]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(AxleBrakesComponent);
    component = fixture.componentInstance;
  });

  it('should create view only with populated data', () => {
    component.axle = TESTING_UTILS.mockAxle();
    component.vehicleType = 'trl';

    fixture.detectChanges();
    expect(component).toBeDefined();
    expect(fixture).toMatchSnapshot();
  });

});
