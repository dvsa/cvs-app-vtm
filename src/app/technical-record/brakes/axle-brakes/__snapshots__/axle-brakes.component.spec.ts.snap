// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AxleBrakesComponent should create view only with populated data 1`] = `
<vtm-axle-brakes
  axle={[Function Object]}
  vehicleType={[Function String]}
>
  <dl
    class="govuk-summary-list"
  >
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        Axle (1)
      </dt>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        Brake actuator
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-brakes-brakeActuator"
      >
         123 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        Lever length
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-brakes-leverLength"
      >
         123 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        Spring brake parking
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-brakes-springBrakeParking"
      >
         Yes 
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
  </dl>
</vtm-axle-brakes>
`;
